[CPU Stream] Add noop for CPU stream record_event() and wait_event()

jvandebon opened the pull request:
Summary: Adds wait_event and record_event endpoints to CPU stream in order to facilitate device-agnostic code. Both methods are noops.

Test Plan: CI

Differential Revision: D68833927




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

Skylion007 commented:
@azzolini Seems like this would fix the issue you had with FSDP2 on CPU

Skylion007 commented:
Fix the trailing whitespace and this looks good.

aazzolini commented:
Great! Now I can use FSDPv2 on CPU :)

jvandebon commented:
@pytorchbot label "topic: not user facing"

wconstab commented:
> Great! Now I can use FSDPv2 on CPU :)

that's pretty exciting, did you confirm its true?  is this just 'functionality' working or does it also do things close to optimally in terms of scheduling copies and comms?  cc @awgu

jvandebon committed:
[CPU Stream] Add noop for CPU stream record_event() and wait_event() (#145935)

Summary:

Adds wait_event and record_event endpoints to CPU stream in order to facilitate device-agnostic code. Both methods are noops.

Test Plan: CI

Differential Revision: D68833927

aazzolini commented:
@wconstab Just functionality -- useful for initial sanity checking sizes larger than number of local GPUs avail.
